compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1079,6 +1079,8 @@ pub enum LocalInfo<'tcx> {
     /// A temporary created during the creation of an aggregate
     /// (e.g. a temporary for `foo` in `MyStruct { my_field: foo }`)
     AggregateTemp,
+    /// A temporary created during the pass `Derefer` to avoid it's retagging
+    Temp,
```

Comment:
Not sure about this name

Reply:
Maybe `DerefTemp`? Just like `AggregateTemp`, but created for references that get immediately dereferenced in place expressions containing dereferences in the middle.

Reply:
Sounds better!